breakout_2/breakout_2_answers.Rmd

```diff
@@ -0,0 +1,78 @@
+raw_dm <- read_xpt(url("https://github.com/phuse-org/TestDataFactory/raw/main/Updated/TDF_SDTM/dm.xpt"))
+raw_ae <- read_xpt(url("https://github.com/phuse-org/TestDataFactory/raw/main/Updated/TDF_SDTM/ae.xpt"))
+```
+
+We'll pick up where we left off from the last exercise with the subsets on the data.
+
+```{r subset}
+dm <- raw_dm %>% 
+  select(USUBJID, STUDYID, SITEID, SUBJID, ARM, AGE, RFSTDTC) %>% 
+  filter(RFSTDTC != "")
+
+total_dm <- dm %>% 
+  mutate(ARM = 'Total')
+
+ae <- raw_ae %>% 
+  select(USUBJID, AESEQ, AETERM, AEBODSYS, AEDECOD, AESTDTC, AEENDTC)
+
+ae_cardiac <- ae %>% 
+  filter(AEBODSYS == 'CARDIAC DISORDERS')
+
+ae_tach <- ae %>% 
+  filter(AEDECOD == 'TACHYCARDIA')
+```
+
+## Bind Rows
+The first way we want to combine data is through stacking them.  This is done using `bind_rows()`.  We have proveded both the basic dm data frame but also a data frame called total_dm that contains all the records from dm but the ARM has been changed to Total for use during summaries where a total group is needed.  Let's combine these two datasets using `bind_rows()` and call the new dataset bind_dm
+
+```{r bindrows}
+bind_dm <- bind_rows(dm, total_dm)
+```
+
+This dataset now contains all observations and variables from the two data frames used as inputs.
+
+## Inner Join
+Another way to combine data is by doing a join.  The first type of join we will be doing is an inner join.  This keeps the data that is found only at the intersection of the two data frames.  
+
+We have provided two data frames for this example based off the overall ae data frame.  ae_cardiac is subset on records with an AEBODSYS of 'CARDIAC DISORDERS' and ae_tach is subset on records with an AEDECOD of 'TACHYCARDIA'.  Perform an inner join on these two datasets using the variables USUBJID and AESEQ.  Assign this new data frame to ae_cardio_tach
+
+```{r innerjoin}
+ae_cardio_tach <- inner_join(ae_cardiac, ae_tach, by = c('USUBJID', 'AESEQ'))
+```
+
+Take a look at the output data frame.  Where are the columns from each of the input data frames?  Notice that any variables that are in both but not used in the `by` argument of `inner_join()` will be suffixed with a .x or .y depending on what dataframe it is from.
+
+## Left Join
+The first type of join we will be doing is a left join.  This keeps the data that is found in the 'left' dataframe and any data from the 'right' data frame that overlaps. 
+
+For this example we will combine the dm and ae data frames.  dm will be our 'left' data frame and ae will be our 'right' data frame with the join being done on the USUBJID variable.  The combined dataset can be assigned to the variable dm_ae
+
+```{r leftjoin}
+dm_ae <- left_join(dm, ae, by = 'USUBJID')
+```
+
+Take a look at the output data frame.  Where are the columns from each of the input data frames?  Find subject '01-701-1033', they don't have any records from ae so they just have the variables from dm populated.
```

Comment:
So for this example, what if instead of demosntrating that it'll leave empty data, what if you left joined DM onto AE (i.e. `left_join(ae, dm, by = 'USUBJID')`. 

You could then state here:
> Notice that subject '01-701-1033' is in the DM dataset, but not in AE. The subject didn't have an Adverse Events, so on they were not included when DM was merged on to DM.

Reply:
I feel like this is a more practical example rather than left joining DM onto AE.  Also in the current iteration of the example it is easy to see that they don't have any AE records versus tracking down a missing subject.